arch/arm/core/aarch64/switch.S

```diff
@@ -33,18 +33,15 @@ GDATA(_k_neg_eagain)
 
 GTEXT(z_arm64_context_switch)
 SECTION_FUNC(TEXT, z_arm64_context_switch)
-#ifdef CONFIG_TRACING
-	stp	xzr, x30, [sp, #-16]!
-	bl	sys_trace_thread_switched_in
-	ldp	xzr, x30, [sp], #16
-#endif
-	/* load _kernel into x1 and current k_thread into x2 */
-	ldr	x1, =_kernel
-	ldr	x2, [x1, #_kernel_offset_to_current]
+	/*
+	 * dummy_thread->switch_handle is 0 when setup by z_cstart()
```

Comment:
This still seems like something that could be fixed without the null check.  Check the way x86_64 does it -- it doesn't load the switch_handle field from a thread, it offsets the switch_handle address it was passed to find the enclosing thread struct.